index/scorch/merge.go

```diff
@@ -59,7 +58,12 @@ OUTER:
 				// lets get started
 				err := s.planMergeAtSnapshot(ourSnapshot, mergePlannerOptions)
 				if err != nil {
-					s.fireAsyncError(fmt.Errorf("merging err: %v", err))
+					if err == ErrClosed {
+						// index has been closed
+						break OUTER
```

Comment:
feels like this is skipping the ourSnapshot.DecRef() at line 67?

Reply:
Ah, yes, good spot @steveyen.